### PR TITLE
fix: externalSchemas problem when using openapi (#691)

### PR DESCRIPTION
**Pull request by imjuni**
* fix cannot pass externalSchemas when using openapi configuration

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by imjuni** [comment]
@Uzlopak Hi, I add new testcase `external schema reference`.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by imjuni** [comment]
is any progress?

File: test/spec/openapi/refs.js [hunk 1]
```diff
@@ -282,3 +282,46 @@ test('uses examples if has property required in body', async (t) => {
...
+          }
+        }
+      }
+    }
+  })
+
+  fastify.get('/', {
+    schema: {
+      response: {
+        200: {
```
Comment (Uzlopak) [inline test/spec/openapi/refs.js:310]: You dont test your change imho. here we need something like { $ref: 'external-schema-01' }

**Review by Uzlopak — CHANGES_REQUESTED** [review]
I am quite hesitant. The PR feels "wrong".
Shouldnt it be also working by using addSchema? Why is there  not a test, that with addSchema also components.schema is set with schemas from addSchema?

Also shouldnt be the test using the "external" schema as reference to prove that it is used internally by this plugin and not only existing in the final openapi definition?

**Pull request closed without merging**